docker/Dockerfile

```diff
@@ -7,16 +7,19 @@ RUN apt-get update && \
 
 WORKDIR /usr/src/app
 
-COPY docker/package.json .
+# install npm packages
+COPY docker/package.json ./
 RUN npm install
 
-# Bundle app source
-COPY build/ ./build
+# copy our code
 COPY docker/server ./server
-COPY start.sh .
+COPY public ./public
+COPY img ./img
+COPY src ./src
+COPY build ./build
```

Comment:
We should probably check for the existence of the `build` directory here and output some guidance to the user if it's missing.

Or we could simplify things even further, and make this kind of check unnecessary.

Our README guides folks to run `npm install && npm run build` in the Quickstart. But on line 12 of this file, we're already running `npm install` for them. We can simplify things by adding a line after that to create the build artifact for them:

```
RUN npm install
RUN npm run build
```

This will guarantee the presence of the `build` directory, so we won't need to check for its existence. We can also then remove the bit from the Quickstart about running `npm install && npm run build`.

Reply:
yeah; let me fix that
- npm install and npm run build will both run during creation of dockerfile. That makes it take longer, but I don't think it's that much of a problem since it only happens first time. And with docker-hub most people won't even need to do that.
- For this to work, I needed to bring in more dependencies into package.json (the docker version) so that it can build. /src is also needed.
- Overall in this way - the build folder is created only on the docker image (although you can also build locally of course) and it's a nice clean way to do it.

Reply:
My 2 cents:

In general, I think having `npm install && npm run build` inside docker file is better. It also means one doesn't need to install all dependencies on their machine.

Reply:
CJ and I went on this back and forth. I also like the fact that inside the docker is cleaner. But CJ made the point that: if you do local dev with the repo, then you will have the build folder locally anyway, so copying it is easier.

Reply:
The other point from CJ was that in this way the whole source code and all packages required to build are not on the docker image as they are not needed for "deployment". we just keep the minimum necessary to deploy, and config/secrets are loaded on run time.